packages/frontend-2/components/invite/dialog/Server.vue

```diff
@@ -133,35 +135,52 @@ const removeInviteItem = (index: number) => {
   removeInvite(index)
 }
 
-const onSubmit = handleSubmit(() => {
+const onSubmit = handleSubmit(async () => {
   const invites = fields.value.filter((invite) => invite.value.email)
 
-  invites.forEach(async (invite) => {
-    invite.value.project
-      ? await inviteUserToProject(invite.value.project.id, [
-          {
-            email: invite.value.email,
-            serverRole: invite.value.serverRole
-          }
-        ])
-      : await inviteUserToServer([
-          {
-            email: invite.value.email,
-            serverRole: invite.value.serverRole
-          }
-        ])
-  })
+  try {
+    await Promise.all(
+      invites.map((invite) =>
+        invite.value.project
+          ? inviteUserToProject(
+              invite.value.project.id,
+              [{ email: invite.value.email, serverRole: invite.value.serverRole }],
+              { hideToasts: true }
+            )
+          : inviteUserToServer(
+              [{ email: invite.value.email, serverRole: invite.value.serverRole }],
+              { hideToasts: true }
+            )
+      )
+    )
 
-  mixpanel.track('Invite Action', {
-    type: 'server invite',
-    name: 'send',
-    multiple: fields.value.length !== 1,
-    count: fields.value.length,
-    hasProject: !!fields.value.some((invite) => invite.value.project),
-    to: 'email'
-  })
+    triggerNotification({
+      type: ToastNotificationType.Success,
+      title:
+        invites.length > 1
+          ? 'Invites successfully send'
+          : `Invite successfully sent to ${invites[0].value.email}`
+    })
 
-  isOpen.value = false
+    mixpanel.track('Invite Action', {
+      type: 'server invite',
+      name: 'send',
+      multiple: fields.value.length !== 1,
+      count: fields.value.length,
+      hasProject: !!fields.value.some((invite) => invite.value.project),
+      to: 'email'
+    })
+
+    isOpen.value = false
+  } catch {
```

Comment:
The change of any of them failing is slim, main cause probably network errors, as validation is already done, but this adds at least some basic error handling in the case this happens